### PR TITLE
Add subscription support

### PR DESCRIPTION
**Pull request by wrandall22**
This PR adds deserialization logic for notifications so it can be in one central place instead of copy/pasted in several other projects.

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>org.ccci.gto</groupId>
     <artifactId>global-registry-client</artifactId>
-    <version>0.2.2-SNAPSHOT</version>
+    <version>0.2.3-SNAPSHOT</version>
```
Comment (frett) [inline api/pom.xml]: shouldn't need to bump the snapshot version, after we merge I'll push a release build

Comment (wrandall22) [inline api/pom.xml]: Okay, I'll remove that commit.

**Comment by wrandall22** [comment]
What will the new version be? I'll need it in the pom for another project I'm working on.

**Comment by frett** [comment]
0.2.2

**Comment by wrandall22** [comment]
Thanks for the review!

**Comment by frett** [comment]
alright, 0.2.2 should be deployed and ready to use

**Comment by wrandall22** [comment]
Maybe I don't have my settings.xml set up right; not finding it. Which repo should have it?

**Comment by frett** [comment]
maven.gcx.org

https://maven.gcx.org/artifactory/simple/public-release/org/ccci/gto/global-registry-client-api/0.2.2/

**Comment by frett** [comment]
arkham should be setup to proxy to maven when necessary. or you could add the following repo config to your maven pom:

```xml
  <repositories>
    <repository>
      <id>maven.gcx.org</id>
      <url>https://maven.gcx.org</url>
    </repository>
  </repositories>
```

**Comment by wrandall22** [comment]
Thanks, all set with it.

**Pull request merged**